main.py

```diff
@@ -464,6 +464,68 @@ def show_ranking(update: Update, context: CallbackContext) -> None:
     )
 
 
+def show_best_messages(update: Update, context: CallbackContext) -> None:
+    days = 7
+    n = 10
+    try:
+        if context.args:
+            days = int(context.args[0])
+            if days < 1:
+                update.message.reply_text("Days argument must be >= 1.")
+                return
+        if len(context.args) > 1:
+            n = min(int(context.args[1]), 30)
+    except ValueError:
+        update.message.reply_text("Usage: /most_reacted <optional days> <optional number of messages> <optional user>")
+        return
+
+    min_timestamp = time.time_ns() - days * (24 * 60 * 60 * 10 ** 9)
+    with get_conn() as conn:
+        if len(context.args) > 2:
+            user = context.args[2]
+            reactions_received = list(
+                conn.execute(
+                    "select message.author_id, message.original_id, count(*) as c from message "
+                    "inner join reaction "
+                    "on message.id = reaction.parent "
+                    "where reaction.timestamp > ? "
+                    "and message.chat_id = ? "
+                    "and message.author = ?"
+                    "group by message.id "
+                    "order by c "
+                    "limit ?",
+                    (
+                        min_timestamp,
+                        update.message.chat_id,
+                        user,
+                        n,
+                    ),
+                ).fetchall()
+            )
+        else:
+            reactions_received = list(
```

Comment:
This query is almost the same as the one above -- could you please refactor it a bit so that there's less duplication?